Authentication/LoginProviders/LoginProvider.swift

```diff
@@ -136,4 +140,14 @@ public protocol LoginProvider {
     */
     func logOut() -> SignalProducer<(), LoginProviderErrorType>
     
+    /**
+        Handles the url returned by the login process. Parameters should be the same of the AppDelegate method.
+    */
+    func handleUrl(_ app: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey : Any]) -> Bool
+    
+    /**
+        Handles the initializing of the providers. Parameters should be the same of the AppDelegate method.
```

Comment:
I would say something like:
```
It takes care of initializing the provider so that it works as expected.
It's used for example for recognizing a user that had already logged in to this app through this provider before.
For this to work correctly, it should receive the same arguments your app's AppDelegate does for this.
Returns true if the provider could be correctly initialized.
```